### ISSUE TITLE
SDL_floor() returns incorrect values

### ISSUE THREAD

**Issue by slouken** [libsdl-org/SDL#5603]
During the N-Gage port, @mupfdev discovered that SDL_floor() was returning incorrect values:
> I was able to isolate the cause of the clipping issue to QueueCmdSetViewport():
> https://github.com/libsdl-org/SDL/blob/f25a78eede5ef7da5f2d45ffc7f9f9e083abd265/src/render/SDL_render.c#L362-L363
> SDL_floor() returns zero, although renderer->viewport.w and renderer->viewport.h have correct values.

We should create math function tests and make sure that SDL's fallback implementations are correct.

**Comment by FtZPetruska** [comment]
Hello, I would be interested in working on these tests.

Before opening a draft PR, I wanted to ensure a couple of things:

- To avoid cluttering the test subfolder, what do you think about placing all the tests into a folder such as`test/libm`, `test/stdinc` or `test/stdlib`?
- Is it enough to test only the `SDL_uclibc_*` functions (defined in `src/libm/math_libm.h`) ? Some functions such as `SDL_ceil` don't have a `SDL_uclib` equivalent and are just defined in place, in order to test all the fallback functions in `SDL_stdinc.h`, there are two solutions I can think of: 
  - make a separate build of SDL with all the `HAVE_<math function>` forcefully undefined
  - extracting the fallback to a separated function, allowing it to be called directly from the test

<details>
<summary> Example of the latter option </summary>

Currently:

```c
double
SDL_ceil(double x)
{
#if defined(HAVE_CEIL)
    return ceil(x);
#else
    double integer = SDL_floor(x);
    double fraction = x - integer;
    if (fraction > 0.0) {
        integer += 1.0;
    }
    return integer;
#endif /* HAVE_CEIL */
}
```

Replaced by:

```c
double
SDL_ceil_fallback(double x)
{
    double integer = SDL_floor(x);
    double fraction = x - integer;
    if (fraction > 0.0) {
        integer += 1.0;
    }
    return integer;
}

double
SDL_ceil(double x)
{
#if defined(HAVE_CEIL)
    return ceil(x);
#else
    return SDL_ceil_fallback(x);
#endif /* HAVE_CEIL */
}
```

Test code:

```c
#include <SDL.h>
#include <SDL_test.h>

/* forward declaration of the fallback function */
double SDL_ceil_fallback(double x);

int main(int argc, char* argv[]) {
    /* ... */
}
```

</details>

**Comment by ccawley2011** [comment]
The issue with `SDL_floor()` may be fixed by PR #5607, although this will need to be confirmed with @mupfdev to be certain.

**Comment by slouken** [comment]
I was thinking we could create testautomation_math.c and add all the math tests there. We don't have to have a special way to test the uclibc functions, we should just test the SDL math functions and they'll apply to however SDL is configured. On Windows SDL is built without libc support so they'll get exercised there.

**Comment by FtZPetruska** [comment]
Sounds good, I will open a draft PR for it shortly.

**Comment by mupfdev** [comment]
> The issue with `SDL_floor()` may be fixed by PR #5607, although this will need to be confirmed with @mupfdev to be certain.

Confirmed. PR #5607 fixes my issue.
